maas/plugins/neutron_metadata_local_check.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+# Copyright 2014, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import shlex
+import subprocess
+
+from maas_common import get_neutron_client
+from maas_common import metric_bool
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+
+# identify the first active neutron agents container on this host
+# network namespaces can only be accessed from within neutron agents container
+FIND_CONTAINER = 'lxc-ls -1 --active .*neutron_agents|head -n 1'
+SERVICE_CHECK = 'ip netns exec %s curl -fvs 127.0.0.1:80'
+
+
+def check(args):
+    network_endpoint = 'http://{host}:9696'.format(host=args.host)
+    try:
+        neutron = get_neutron_client(endpoint_url=network_endpoint)
+
+    # not gathering api status metric here so catch any exception
+    except Exception as e:
+        status_err(str(e))
+
+    # gather networks to check
+    # only check networks which have a port, otherwise the proxy won't answer
+    ports = neutron.list_ports()['ports']
+    target_nets = [p['network_id'] for p in ports]
+    nets = [n for n in neutron.list_networks()['networks']
+            if n['id'] in target_nets]
```

Comment:
this can be simplified: eliminate list_networks() call since I am only using the network_id.